action/protocol/rewarding/reward.go

```diff
@@ -215,7 +215,8 @@ func (p *Protocol) GrantEpochReward(
 	}
 
 	// Reward additional bootstrap bonus
-	if epochNum <= a.foundationBonusLastEpoch {
+	fairBankEpochNum := rp.GetEpochNum(hu.FairbankBlockHeight()) // extend foundation bonus from fairbank to fairbank + 1 year
+	if epochNum <= a.foundationBonusLastEpoch || (epochNum >= fairBankEpochNum && epochNum <= fairBankEpochNum+a.foundationBonusLastEpoch) {
```

Comment:
should calculate numEpochIn1Year = xxx
and <= fairBankEpochNum+numEpochIn1Year

a.foundationBonusLastEpoch just happens to be equal to 1 year now, but should not count on that

Reply:
yes, ```numEpochIn1Year``` should be in genesis? or const variable?

Reply:
i think it is more clear to have a new start and new end to indicate this is the second wave of foundation bonus.
I think an end time is better than a start time + a last period.  the reason is if we want to extend that end time, it is easier. if we use `numEpochIn1Year`, we may introduce a bug by changing that

Reply:
yes, I changed to add that 2 fields(start, end epoch) in admin struct in the last commit. 
But still we need to specify ```numEpochIn1Year``` to set up end epoch value, however in the future, we always can change it I guess

Reply:
we can calculate end epoch offline